STUDIO/org.openl.rules.repository/src/org/openl/rules/repository/folder/CombinedFileChanges.java

```diff
@@ -0,0 +1,52 @@
+package org.openl.rules.repository.folder;
+
+import java.util.Collection;
+import java.util.Iterator;
+
+import org.openl.rules.repository.api.FileItem;
+import org.openl.util.IOUtils;
+
+public class CombinedFileChanges implements Iterable<FileItem> {
+    private final Collection<Iterable<FileItem>> multipleIterables;
+
+    public CombinedFileChanges(Collection<Iterable<FileItem>> multipleIterables) {
+        this.multipleIterables = multipleIterables;
+    }
+
+    @Override
+    public Iterator<FileItem> iterator() {
+        Iterator<Iterable<FileItem>> iterableIterator = multipleIterables.iterator();
+        return new Iterator<FileItem>() {
+            private Iterable<FileItem> fileItemIterable;
+            private Iterator<FileItem> fileItemIterator;
+
+            @Override
+            public boolean hasNext() {
+                if (fileItemIterator != null && fileItemIterator.hasNext()) {

```

Comment:
seems like fileItemIterator cannot be null

Reply:
It can. In the beginning of iteration it's null.